src/webview/connections.ts

```diff
@@ -0,0 +1,139 @@
+        break;
+      }
+      case 'deleteConnection': {
+        deleteConnection();
+        break;
+      }
+      case 'connections.refreshConnections': {
+        refreshConnections();
+        break;
+      }
+      case 'connections.createConnectionsView': {
+        createConnectionsView(message.connections, message.currentConnection);
+        break;
+      }
+    }
+  });
+
+  function updateConnectionList(connections, currentConnection) {
+    const connectionsList = document.querySelector('.connections-list');
+    connectionsList.textContent = 'You have no saved connections. Add one below.';
+    if (connections !== undefined) {
+      connectionsList.textContent = '';
+      for (key in connections) {
+        if (connections.hasOwnProperty(key)) {
+          let connection = connections[key];
+          let connectionName = connection['name'];
+          let connectionURL = connection['url'];
+
+          let div = document.createElement('div');
+          div.className = connectionName;
+          let connectionNameLabel = document.createElement('h3');
+          connectionNameLabel.className = 'ConnectionNameLabel';
+          connectionNameLabel.innerHTML = 'Connection: ' + connectionName;
+
+          div.className = connectionURL;
+          let connectionURLLabel = document.createElement('label');
+          connectionURLLabel.className = 'ConnectionURLLabel';
+          connectionURLLabel.innerHTML = connectionURL;
+
+          div.appendChild(connectionNameLabel);
+          div.appendChild(document.createElement('br'));
+          div.appendChild(connectionURLLabel);
+          div.appendChild(document.createElement('br'));
+          div.appendChild(document.createElement('br'));
+
+          let deleteButton = document.createElement('button');
+          deleteButton.className = 'DeleteButton';
+          deleteButton.id = 'Delete-' + connectionName;
+          deleteButton.innerHTML = 'Delete';
+
+          deleteButton.onclick = function () {
+            deleteConnection(connectionName);
+          };
+          div.appendChild(deleteButton);
+
+          div.appendChild(document.createElement('br'));
+          div.appendChild(document.createElement('br'));
+
+          let updateButton = document.createElement('button');
+          updateButton.className = 'UpdateButton';
+          updateButton.id = 'Update-' + connectionName;
+          updateButton.innerHTML = 'Update';
+
+          updateButton.onclick = function () {
+            updateConnection(connectionName);
+          };
+          div.appendChild(updateButton);
+
+          div.appendChild(document.createElement('br'));
+          div.appendChild(document.createElement('br'));
+
+          let connectButton = document.createElement('button');
+          connectButton.className = 'ConnectButton';
+          connectButton.id = 'Connect-' + connectionName;
+
+          if (connectionName === currentConnection['name']) {
```

Comment:
This currently causes the handler to crash if there is no currently active connection.